devimint/src/federation.rs

```diff
@@ -281,8 +296,9 @@ impl Federation {
     }
 
     pub async fn generate_epochs(&self, epochs: usize) -> Result<()> {
+        let finality_delay = self.get_finality_delay().await?;
         for _ in 0..epochs {
-            self.bitcoind.mine_blocks(10).await?;
+            self.bitcoind.mine_blocks(finality_delay.into()).await?;
```

Comment:
Mining finality delay many block isn't necessary each time to generate an epoch, I guess it helps with the first time that wouldn't trigger if height-delay<0.